nixos/modules/services/monitoring/chronograf.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.chronograf;
+in
+{
+
+  options.services.chronograf = {
+    enable = mkEnableOption "chronograf server";
+
+    port = mkOption {
+      default = 8888;
+      type = types.int;
+      description = ''
+        The port where chronograf is going to be listening.
+      '';
+    };
+
+    dataDir = mkOption {
+      default = "/var/db/chronograf";
+      type = types.str;
+      description = ''
+        Data directory for chronograf data files.
+      '';
+    };
+
+    influxdb = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        The URL of the InfluxDB that Chronograf is going to connect to.
+      '';
+      example ="http://localhost:8086";
+    };
+
+    kapacitor = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        The URL of the Kapacitor that Chronograf is going to connect to.
+      '';
+    };
+
+    reporting = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Disable reporting of usage stats (os,arch,version,cluster_id,uptime) once every 24hr.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.chronograf ];
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0770 chronograf chronograf - -"
+    ];
+
+    systemd.services.chronograf = {
+      description = "Chronograf server";
+      after = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = ''${pkgs.chronograf}/bin/chronograf --port=${toString cfg.port} --bolt-path=${cfg.dataDir}/chronograf-v1.db ${if cfg.influxdb != "" then "--influxdb-url=${cfg.influxdb}" else ""} ${if cfg.kapacitor != "" then "--kapacitor-url=${cfg.kapacitor}" else ""} ${if !cfg.reporting then "--reporting-disabled" else ""}'';
```

Comment:
It would be nice if you could split this over multiple lines for easier reading.